### PR TITLE
Handle multiple snapshots (creation + deletion) at the same time

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: ovs/lib/vdisk.py [hunk 1]
```diff
@@ -350,15 +380,39 @@ def delete_snapshot(vdisk_guid, snapshot_id):
...
+        vdisk_result = result[vdisk_guid]
+        if vdisk_result[0] is False:
+            raise RuntimeError(vdisk_result[1])
+
+    @staticmethod
+    @celery.task(name='ovs.vdisk.delete_snapshots')
+    def delete_snapshots(snapshot_mapping):
+        """
+        Delete vDisk snapshots
+        :param snapshot_mapping: Mapping of VDisk guid and snapshot_id
```
Comment (kvanhijf) [inline ovs/lib/vdisk.py]: `snapshot_id` is the internal codename and should not be reflected in the docstrings

File: ovs/lib/vdisk.py [hunk 2]
```diff
@@ -350,15 +380,39 @@ def delete_snapshot(vdisk_guid, snapshot_id):
...
+        :type snapshot_mapping: dict
+        """
+        results = {}
+        for vdisk_guid, snapshot_id in snapshot_mapping.iteritems():
+            try:
+                vdisk = VDisk(vdisk_guid)
+                if snapshot_id not in [snap['guid'] for snap in vdisk.snapshots]:
```
Comment (kvanhijf) [inline ovs/lib/vdisk.py:396]: Shouldn't we invalidate the snapshots at this point?

File: ovs/lib/vdisk.py [hunk 3]
```diff
@@ -323,22 +323,52 @@ def create_snapshot(vdisk_guid, metadata):
...
-                                                   snapshot_id=str(snapshot_id),
-                                                   metadata=metadata)
-        vdisk.invalidate_dynamics(['snapshots'])
-        return snapshot_id
+        results = {}
+        for guid in vdisk_guids:
+            try:
+                vdisk = VDisk(guid)
+            except ObjectNotFoundException:
+                results[guid] = [False, 'VDisk could not be found'.format(guid)]
```
Comment (kvanhijf) [inline ovs/lib/vdisk.py]: Incorrect string formatting

File: webapps/api/backend/views/vpools.py [hunk 4]
```diff
@@ -180,3 +181,40 @@ def devicename_exists(self, vpool, name=None, names=None):
...
         return False
+
+    @action()
+    @log()
+    @required_roles(['read', 'write'])
+    @return_task()
+    @load(VPool)
+    def create_snapshots(self, vdisk_guids, name, timestamp=None, consistent=False, automatic=False, sticky=False):
+        """
+        Creates a snapshot from the vDisk
```
Comment (kvanhijf) [inline webapps/api/backend/views/vpools.py]: Docstring incorrect

File: webapps/api/backend/views/vpools.py [hunk 5]
```diff
@@ -180,3 +181,40 @@ def devicename_exists(self, vpool, name=None, names=None):
...
+        :param vdisk_guids: Guids of the virtual disk to create snapshot from
+        :param name: Name of the snapshot (label)
+        :param timestamp: Timestamp of the snapshot - integer
```
Comment (kvanhijf) [inline webapps/api/backend/views/vpools.py]: Add `:param type:` instead please

File: ovs/lib/vdisk.py (hunk 1 above)
Comment (khenderick) [reply ovs/lib/vdisk.py]: Correct

**Pull request merged**